### PR TITLE
feat:added plugins chips filter in microsite

### LINKED ISSUE #13014: [microsite] Plugin Marketplace - Category Filter
## Feature Suggestion

It would be great now that there are many items in the Plugin Marketplace that we add a filter by category so that way you can find plugins a little more easily

## Possible Implementation

This could simply be a drop down with all the available categories or as complex as a multi-select and then the cards would filter based on the selection

## Context

Now that there are lots of awesome plugins it's getting a bit harder to find things

### PR DESCRIPTION
**Pull request by antoniobergas**
## Hey, I just made a Pull Request!

New chips filter by category for Plugins page in the microsite to make easier to find out the plugnins you are looking for

image

image

#### :heavy_check_mark: Checklist

- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Closes #13014

### CODE REVIEW

**Comment by antoniobergas** [comment]
Hi! still interested on have some feedback on this 🤗

File: microsite/package.json
Comment (freben) [inline microsite/package.json:37]: I would contest that adding the entire MUI and emotion here, is overkill. We've been holding off on doing that intentionally. The infima package that comes builtin with docusaurus, [has pills](https://infima.dev/docs/components/pills/) that we probably should use instead.

**Comment by awanlin** [comment]
Hi @antoniobergas, sorry for the slow follow up. I really appreciate the work you've done I just feel like this design/implementation puts a lot in front of the user at once where as a proper drop down would be much cleaner. I also think the list could get very long which means this could get so big that we don't even see the actual content till we scroll.

Sorry, not trying to diminish the work you've done

File: microsite/package.json
Comment (antoniobergas) [reply microsite/package.json:37]: Thanks for the feedback, that make sense! I missed the infima integration, will make the changes and remove the material dependencies!

**Comment by antoniobergas** [comment]
Hi @awanlin, yes that make totally sense I didn't think in that. So I will back to the multi dropdown. Any suggestions for show which categories you have selected?

What I can do maybe it's combine it, so the dropdown at the right side and when you select any category the chip shown at the left? 

Does that make sense from a user perspective?

**Comment by awanlin** [comment]
Sorry @antoniobergas, I should have shared what I was thinking about this. My thought was to have something a lot like what we already see in Backstage itself. So like the way the OwnerPicker looks and works:

image

But this is much easier said then done seeing as you have to figure out how to do that with Infima

**Comment by antoniobergas** [comment]
No problem 😄 thanks for the sharing, now that the base is more or less done I will see what I can get to show properly to the user 👌

**Comment by antoniobergas** [comment]
I changed the chips with a multi dropdown. I think the result is good 😄

image

Unfortunately I have a weird issue with the styles. First time I load the page with yarn start the label styles are overrited and after refresh they have the custom properties I used

image

image


Does anyone have an idea why is this happening?

**Comment by antoniobergas** [comment]
After some tests removing infima dropdown classes and my custom scss still having the issue.

I will continue debugging, any suggestions will be apreciatted, have not much experience with docosaurus 😅

**Comment by antoniobergas** [comment]
At the end was the label tag for some reason, after change it to a div everything works fine 😄

File: microsite/src/pages/plugins/plugins.module.scss
Comment (awanlin) [inline microsite/src/pages/plugins/plugins.module.scss:31]: My CSS isn't the best but I worry about changing this from the fraction to something more rigid. Might that cause issues art different screen sizes?

**Review by awanlin — CHANGES_REQUESTED** [review]
Looks good to me, left one comment but will leave this to the maintainers to do a more detailed review

File: microsite/src/pages/plugins/plugins.module.scss
Comment (antoniobergas) [reply microsite/src/pages/plugins/plugins.module.scss:31]: I test a bit with different resolutions and in mobile view on the browser and in theory not. 

I changed that becase the 1fr make the cards wider when you filter them and leave for example 1 or 2 cards only 😅

My css isnt also the best so any suggestion to maintain the cards with the same width at filter will be apreciatted

**Comment by antoniobergas** [comment]
Hey is there a way I can re-run the build? I tested locally with node 16 and works fine

**Comment by awanlin** [comment]
> Hey is there a way I can re-run the build? I tested locally with node 16 and works fine

I'll trigger it to see but I think it might be due to this: https://github.com/backstage/backstage/pull/17549

**Comment by antoniobergas** [comment]
Thanks for pointing me in the right direction! 🤗

**Pull request closed without merging**
